apps/src/templates/certificates/Certificate.jsx

```diff
@@ -65,7 +65,15 @@ function Certificate(props) {
       course: props.tutorial,
       donor,
     };
-    return btoa(reEncodeNonLatin1(JSON.stringify(data)));
+    const asciiData = btoa(reEncodeNonLatin1(JSON.stringify(data)));
+    const urlSafeData = asciiData
+      .replace(/\+/g, '-')
```

Comment:
Consider using `.replaceAll()` which replaces every occurrence in the string, vs. only the first. This would make it so you don't need to use regex with the global modifier.

Example: `.replaceAll('+', '-')`

Reply:
We can't used replaceAll. The testing environment to fails because the replaceAll() method was introduced in ECMAScript 2021, so if the JavaScript environment being used for the unit tests is using an older version of JavaScript is being used, the original code could throw an error.

The previous PR ran into this issue, and luckily @mikeharv had some ideas. See this PR [#51770](https://github.com/code-dot-org/code-dot-org/pull/51770)

Reply:
Ah I see. Bummer, hopefully we can get the JS version updated soon.

Reply:
Lol you might want to leave a comment since future engineers will also think "why isn't this using replaceAll?" and link to something explaining why